### ISSUE TITLE
Adding center pairs (ii-pairs)

### ISSUE THREAD

**Issue by mastricker** [cosmo-epfl/librascal#104]
Felix and I have tried to implement this as a child of `adapptor_filter`. The big issue is (Felix has forseen that), that is is designed to make a smaller list, not add atoms. Even if the cluster ref of an ii-pair is constructed correctly. All sorts of layer-issues arise.
I am now in the process of adding the functionality to `adaptor_neighbour_list` as a flag (like considering ghost neighbours).
This seems, after all, the simpler and cleaner way.

The big issue is that adding pairs and fiddling with the `ClusterLayer` apparently leads to a very weird neighbour list: pairs which all belong to one atom are separated out and appear to be treated differently.

And a question arises:
If one includes ii-pairs. Are they to be retained in the increase maxorder functionality? That would also need changes in `adaptor_increase_maxorder`.

**Comment by mastricker** [comment]
On the way there: I have encountered several test for neighbour lists, which did not actually initialize the neighbourhood. I.e.: number of pairs was 0, compared to a counter. No pairs.
Corrected that.

**Comment by mastricker** [comment]
Missing now is the binding part.

**Comment by felixmusil** [comment]
The downside of integrating it to the adaptor neighbor list is that the manager lammps misses this functionality too. I don't think it's a big issue to add it separately though.

Regarding the max order, does it touch at the order 1 and 2 part of the neighbors list ? If yes, why does it have to ? 
I have the feeling it shouldn't keep this feature since the iik triplet makes little sense but it might still be an entry of the descriptor ?

**Comment by mastricker** [comment]
Ok, I did not think of that - but I remember we talked about it last week when we met.

We might still need to go for a barebones adaptor to just add the pairs. I mean one solution is to rebuild the neighbourlist when using lammps, but that is just waste of computation time.
I'll think about it. Probably make a new, independent adaptor.

We could nevertheless keep the option in the neighbour list adaptor for cases when that is the only neighbour list. And we use the _other one_ when dealing with existing neighbour lists.

How do your ipi and quippy things (if these are all) work with librascal? Are there neighbourlists already?
